src/cnd/cnd.ts

```diff
@@ -259,15 +259,27 @@ export class Cnd {
     );
   }
 
+  /**
+   * Get the peer id of the cnd node
+   *
+   * @returns Promise that resolves with the peer id of the cnd node,
+   * @throws A {@link Problem} from the cnd REST API or an Error.
+   */
   public async getPeerId(): Promise<string> {
     const info = await this.getInfo();
     if (!info.id) {
-      throw new Error("id field not present");
+      return Promise.reject(new Error("id field not present"));
```

Comment:
I did a quick google on this but didn't understand it :D 
Why are you using `Promise.reject` here and not `throw new Error`?

Reply:
It's an artefact from when I was playign with `reject`/`resolve` to try to implement Yos' proposal.